framework/source/class/qx/ui/core/queue/Appearance.js

```diff
@@ -88,6 +94,7 @@ qx.Class.define("qx.ui.core.queue.Appearance",
         // Order is important to allow the same widget to be re-queued directly
         obj = queue[i];
         queue.splice(i, 1);
+        delete this.__lookup[obj.$$hash]
```

Comment:
Also the comment above the array editing was part of the original source and says that objects can be re-queued, which is why the array is edited rather than being reset to [] at the end of the loop.